### PR TITLE
feat(ecr): support imageScanOnPush when creating the repository

### LINKED ISSUE #7471: enable imageScanOnPush when creating ECR repo
Amazon ECR has supported the `imageScanOnPush` capabilities, however, due to the lack of cloudformation support(see https://github.com/aws/containers-roadmap/issues/552 and https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/245), we can't simply enable this feature when we create the `Repository` resource in AWS CDK aws-ecr construct.

### Use Case

To create the ECR repository and enable the `scanOnPush` configuration at the same time.

### Proposed Solution

to support it with `AwsCustomResource`

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by pahud**
- [x] implementation
- [x] integ test
- [x] unit test
- [x] 100% test coverage for this PR
- [x] update README

### Commit Message
feat(ecr): support imageScanOnPush when creating the repository

Add a new imageScanOnPush property for the new repository creation.

Fix #7471 
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecr/lib/repository.ts [hunk 1]
```diff
@@ -422,6 +430,36 @@ export class Repository extends RepositoryBase {
...
+          action: 'putImageScanningConfiguration',
+          parameters: {
+            repositoryName: this.repositoryName,
+            imageScanningConfiguration: {
+              scanOnPush: false,
+            },
+          },
+          physicalResourceId: cr.PhysicalResourceId.of(this.repositoryName),
+        },
+        policy: cr.AwsCustomResourcePolicy.fromSdkCalls({ resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE }),
```
Comment (jogold) [inline packages/@aws-cdk/aws-ecr/lib/repository.ts:460]: Shouldn't this be restricted to `this.repositoryArn`?

**Comment by pahud** [comment]
Hi @jogold @MrArnoldPalmer 

I am ready to review now.

File: packages/@aws-cdk/aws-ecr/lib/repository.ts (hunk 1 above)
Comment (jogold) [reply packages/@aws-cdk/aws-ecr/lib/repository.ts:460]: You marked this as resolved but didn't change it?

File: packages/@aws-cdk/aws-ecr/lib/repository.ts [hunk 2]
```diff
@@ -422,6 +430,36 @@ export class Repository extends RepositoryBase {
...
+        onDelete: {
+          service: 'ECR',
+          action: 'putImageScanningConfiguration',
+          parameters: {
+            repositoryName: this.repositoryName,
+            imageScanningConfiguration: {
+              scanOnPush: false,
+            },
+          },
+          physicalResourceId: cr.PhysicalResourceId.of(this.repositoryArn),
```
Comment (pahud) [inline packages/@aws-cdk/aws-ecr/lib/repository.ts:458]: Hi @jogold  it's restricted to `this.repositoryArn` now.

File: packages/@aws-cdk/aws-ecr/lib/repository.ts (hunk 1 above)
Comment (pahud) [reply packages/@aws-cdk/aws-ecr/lib/repository.ts:460]: Hi @jogold 

I did.  Did I miss anything?

https://github.com/aws/aws-cdk/blob/18dec7845a697139c368f27db0b7b077e6c7a6e6/packages/%40aws-cdk/aws-ecr/lib/repository.ts#L447-L458

File: packages/@aws-cdk/aws-ecr/lib/repository.ts (hunk 2 above)
Comment (jogold) [reply packages/@aws-cdk/aws-ecr/lib/repository.ts:458]: No it's not, have a look at https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/custom-resources#execution-policy-1

Comment (pahud) [reply packages/@aws-cdk/aws-ecr/lib/repository.ts:458]: @jogold Thanks for the heads-up. Just fixed it. Please review again.

**Comment by pahud** [comment]
copy @eladb 

It's been 6 days since last review. Would be appreciated if someone can review it again.

thanks

**Pull request merged**